src/console/clients/udp/responses.rs

```diff
@@ -1,9 +1,65 @@
 //! Aquatic responses are not serializable. These are the serializable wrappers.
 use std::net::{Ipv4Addr, Ipv6Addr};
 
-use aquatic_udp_protocol::{AnnounceResponse, Ipv4AddrBytes, Ipv6AddrBytes, ScrapeResponse};
+use anyhow::Context;
+use aquatic_udp_protocol::Response::{self};
+use aquatic_udp_protocol::{AnnounceResponse, ConnectResponse, ErrorResponse, Ipv4AddrBytes, Ipv6AddrBytes, ScrapeResponse};
 use serde::Serialize;
 
+pub trait DtoToJson {
+    /// # Errors
+    ///
+    /// Will return an error if serialization fails.
+    ///
+    fn print_response(&self) -> anyhow::Result<()>
+    where
+        Self: Serialize,
+    {
+        let pretty_json = serde_json::to_string_pretty(self).context("response JSON serialization")?;
+        println!("{pretty_json}");
```

Comment:
Hi @mario-nt I would remove the `println` from the trait. I would return a `String`. That would make it easier to test the implementations and you can use the method for other cases. Maybe you want to serialize to JSON but you don't want to print it in many cases.